### PR TITLE
3.10 Changed content to align with the Aer Simulator Update

### LINKED ISSUE #1191: [3.10] Problems in simulator used in  content and code
**Describe the issue**
In this section of textbook clearly two simulation methods are used Statevector and Qasm . But there is a problem in content as these simulation method are part of `aer_simulator`. One can clearly see that only `aer_simulator` is defined in code but the statevector and qasm are used in content.  

**Describe what you expect to see**
I think one should fix the content and change all the simulators to `aer`  simulation methods and make it compatible with statevector and qasm

### PR DESCRIPTION
**Pull request by tgag17**
# Changes made

Changed headings and content from the `Statevector` and `QASM` simulator to `Aer` simulator.
# Justification

The code uses only the aer simulator while the content indicated the use of the statevector and qasm simulator. 
Fixes #1191

### CODE REVIEW

**Comment by divshacker** [comment]
Very Well changed! Great work @tgag17

File: content/ch-algorithms/teleportation.ipynb [hunk 1]
```diff
@@ -6479,16 +6479,16 @@
...
-    "(`Initialize` is technically not a gate since it contains a reset operation, and so is not reversible. We call it an 'instruction' instead). If the quantum teleportation circuit works, then at the end of the circuit the qubit $|q_2\\rangle$ will be in this state. We will check this using the statevector simulator."
+    "(`Initialize` is technically not a gate since it contains a reset operation, and so is not reversible. We call it an 'instruction' instead). If the quantum teleportation circuit works, then at the end of the circuit the qubit $|q_2\\rangle$ will be in this state. We will check this using the aer simulator."
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### 3.2 Using the Statevector Simulator <a id='simulating-sv'></a>\n",
+    "### 3.2 Using the Final Statevector <a id='simulating-sv'></a>\n",
```
Comment (frankharkins) [inline content/ch-algorithms/teleportation.ipynb:6489]:
```suggestion
    "### 3.2 Using the Simulated Statevector <a id='simulating-sv'></a>\n",
```
I think we should mention there is a simulation happening in the heading.

File: content/ch-algorithms/teleportation.ipynb [hunk 2]
```diff
@@ -18985,9 +18985,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### 3.3 Using the QASM Simulator <a id='simulating-qs'></a>\n",
+    "### 3.3 Using the Final Counts <a id='simulating-fc'></a>\n",
```
Comment (frankharkins) [inline content/ch-algorithms/teleportation.ipynb:18988]:
```suggestion
    "### 3.3 Using the Simulated Counts <a id='simulating-fc'></a>\n",
```
Same here

**Review by frankharkins — DISMISSED** [review]
I would prefer slightly different wording for the headings, but if you disagree I won't block merging. Otherwise looks good. Thanks!

File: content/ch-algorithms/teleportation.ipynb [hunk 3]
```diff
@@ -15,16 +15,16 @@
...
     "## Contents\n",
     "\n",
     "1. [Overview](#overview)    \n",
     "2. [The Quantum Teleportation Protocol](#how)       \n",
     "3. [Simulating the Teleportation Protocol](#simulating)   \n",
     "    3.1 [How will we Test this Result on a Real Quantum Computer?](#testing)   \n",
-    "    3.2 [Using the Statevector Simulator](#simulating-sv)     \n",
-    "    3.3 [Using the QASM Simulator](#simulating-qs)  \n",
+    "    3.2 [Using the Final Statevector](#simulating-sv)     \n",
+    "    3.3 [Using the Final Counts](#simulating-fc)  \n",
```
Comment (divshacker) [inline content/ch-algorithms/teleportation.ipynb:27]:
```suggestion
    "    3.2 [Using the Simulated Statevector](#simulating-sv)     \n",
    "    3.3 [Using the Simulated Counts](#simulating-fc)  \n",
```

Comment (divshacker) [reply content/ch-algorithms/teleportation.ipynb:27]: This is also left @frankharkins

**Review by divshacker — COMMENTED** [review]
I agree with @frankharkins the final is kinda misleading. I think simulated fits best.

**Comment by tgag17** [comment]
Thank you @frankharkins and @divshacker. I agree, 'Simulated' is a better word for the heading. I have made the required commits.

**Review by frankharkins — APPROVED** [review]
Thank you!

**Pull request merged**